src/main/java/com/hubspot/jinjava/el/tree/impl/Scanner.java

```diff
@@ -0,0 +1,453 @@
+        }
+        Symbol symbol = Symbol.INTEGER;
+        if (i < l && input.charAt(i) == '.') {
+            i++;
+            while (i < l && isDigit(input.charAt(i))) {
+                i++;
+            }
+            symbol = Symbol.FLOAT;
+        }
+        if (i < l && (input.charAt(i) == 'e' || input.charAt(i) == 'E')) {
+            int e = i;
+            i++;
+            if (i < l && (input.charAt(i) == '+' || input.charAt(i) == '-')) {
+                i++;
+            }
+            if (i < l && isDigit(input.charAt(i))) {
+                i++;
+                while (i < l && isDigit(input.charAt(i))) {
+                    i++;
+                }
+                symbol = Symbol.FLOAT;
+            } else {
+                i = e;
+            }
+        }
+        return token(symbol, input.substring(position, i), i - position);
+    }
+
+    /**
+     * token inside an eval expression
+     */
+    protected Token nextEval() throws ScanException {
+        char c1 = input.charAt(position);
+        char c2 = position < input.length()-1 ? input.charAt(position+1) : (char)0;
+
+        switch (c1) {
+            case '*': return fixed(Symbol.MUL);
+            case '/': return fixed(Symbol.DIV);
+            case '%': return fixed(Symbol.MOD);
+            case '+': return fixed(Symbol.PLUS);
+            case '-': return fixed(Symbol.MINUS);
+            case '?': return fixed(Symbol.QUESTION);
+            case ':': return fixed(Symbol.COLON);
+            case '[': return fixed(Symbol.LBRACK);
+            case ']': return fixed(Symbol.RBRACK);
+            case '(': return fixed(Symbol.LPAREN);
+            case ')': return fixed(Symbol.RPAREN);
+            case ',': return fixed(Symbol.COMMA);
+            case '.':
+                if (!isDigit(c2)) {
+                    return fixed(Symbol.DOT);
+                }
+                break;
+            case '=':
+                if (c2 == '=') {
+                    return fixed(Symbol.EQ);
+                }
+                break;
+            case '&':
+                if (c2 == '&') {
+                    return fixed(Symbol.AND);
+                }
+                break;
+            case '|':
+                if (c2 == '|') {
+                    return fixed(Symbol.OR);
+                }
+                break;
+            case '!':
+                if (c2 == '=') {
+                    return fixed(Symbol.NE);
+                }
+                return fixed(Symbol.NOT);
+            case '<':
+                if (c2 == '=') {
+                    return fixed(Symbol.LE);
+                }
+                return fixed(Symbol.LT);
+            case '>':
+                if (c2 == '=') {
+                    return fixed(Symbol.GE);
+                }
+                return fixed(Symbol.GT);
+            case '"':
+            case '\'': return nextString();
+            default: throw new IllegalArgumentException("Value " + c1 + " is not supported.");
```

Comment:
This here causes a very large amount of the unit tests to fail. As this PR is currently, over half of the unit tests are failing.

Reply:
Sorry, just realised you had marked this as a draft PR 3 days ago, I had originally looked over it when it was first opened.

Reply:
> This here causes a very large amount of the unit tests to fail.

Fixed))